docs/upgrade/puppet.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: documentation
+title: Puppet Upgrade
+sidebar: sidebars/documentation.html
+script: osmenu.js
+---
+
+# Puppet Upgrade
+
+This document outlines how to upgrade from Puppet 3 to Puppet 4.
+
+##  Overview
+
+Puppet 4 support was added to Foreman in version 1.12, and in Katello in 3.2.
+Puppet 4 is a major upgrade from Puppet 3; the changes are too numerous for
+this document. More information can be found in the [Puppet 4.0 release
+notes](https://docs.puppet.com/puppet/4.0/reference/release_notes.html) and the
+[Puppet upgrade
+prerequsites](https://docs.puppet.com/puppet/latest/reference/upgrade_major_pre.html).
```

Comment:
I might switch the order of these links, and the text of this one to match the page's title  - it's probably the most important thing users to need to read.